libs/wallet/src/web3-react/Web3Provider/index.tsx

```diff
@@ -1,16 +1,27 @@
 import { ReactNode, useMemo } from 'react'
 
-import { getConnectionName, Web3ReactConnection } from '@cowprotocol/wallet'
 import { Web3ReactHooks, Web3ReactProvider } from '@web3-react/core'
 import { Connector } from '@web3-react/types'
 
 import { useEagerlyConnect } from './hooks/useEagerlyConnect'
 import { useOrderedConnections } from './hooks/useOrderedConnections'
 
-export default function Web3Provider({ children }: { children: ReactNode }) {
-  useEagerlyConnect()
+import { ConnectionType } from '../../api/types'
+import { getConnectionName } from '../../api/utils/connection'
+import { Web3ReactConnection } from '../types'
 
-  const connections = useOrderedConnections()
+export function Web3Provider({
+  children,
+  selectedWallet,
+  selectedWalletBackfilled,
+}: {
+  children: ReactNode
+  selectedWallet: ConnectionType | undefined
+  selectedWalletBackfilled: boolean
```

Comment:
nit: declare and export the type for the params object

Reply:
Fixed.
Added a link to the params description: https://github.com/cowprotocol/cowswap/pull/4414/commits/38c0ad33bbf935db2f7fe4528e369a18df31de63